src/data.jl

```diff
@@ -907,23 +954,23 @@ function jlconvert(::ReadRepresentation{T,DataTypeODR()}, f::JLDFile,
         end for ref in paramrefs]
     end
 
-    path = String(jlconvert(ReadRepresentation{UInt8,Vlen{UInt8}}(), f, ptr, NULL_REFERENCE))
-    parts = split(path, '.')
-    sym = Symbol(parts[1])
-    m = filter(m->nameof(m) == sym, keys(Base.module_keys))
-    if isempty(m)
-        return hasparams ? UnknownType(path, params) : UnknownType(path)
-    end
-    m = pop!(m)
-    for part in parts[2:end]
-        sym = Symbol(part)
-        if !isdefined(m, sym)
-            return hasparams ? UnknownType(path, params) : UnknownType(path)
-        end
-        m = getfield(m, sym)
-    end
-    if !isa(m, DataType) && !isa(m, UnionAll)
-        return hasparams ? UnknownType(path, params) : UnknownType(path)
+    mypath = String(jlconvert(ReadRepresentation{UInt8,Vlen{UInt8}}(), f, ptr, NULL_REFERENCE))
```

Comment:
Since all but the last argument are identical, maybe this would be slightly simpler as
```julia
m = _resolve_type(rr, f, ptr, header_offset, mypath, hasparams, hasparams ? params : nothing)
```
?

Reply:
I agree. Done.